### PR TITLE
Allow detecting tokens for different blockchains using the API

### PR DESCRIPTION
**Pull request by PabloCastellano**
This is necessary in order to support other blockchains and L2s in rotki

Changes:

1. Generalized endpoint /api/(version)/blockchains/ETH/tokens/detect to /api/(version)/blockchains/(blockchain)/tokens/detect
2. Renamed and generalized methods such as detect_ethereum_tokens(), query_ethereum_tokens() and _add_protocol_balances() to allow other EVMs

Related: #4725

Requires: ~#4806~ ~#4807~ ~#4829~

### CODE REVIEW

**Comment by PabloCastellano** [comment]
Open question: since EthTokens become Evmtokens, should I also move this class somewhere outside of `rotkehlchen/chain/ethereum/tokens.py`?

**Comment by LefterisJP** [comment]
> Open question: since EthTokens become Evmtokens, should I also move this class somewhere outside of `rotkehlchen/chain/ethereum/tokens.py`?

Hey @PabloCastellano about this PR, lots of changes happening at once :sweat_smile: 

So yeah a new location would make sense but for this to really work and make sense it would need also a chain agnostic version of `EthereumManager` which would be the `EVMManager`.

To have a chain agnostic `EVMManager` I have not really thought what is needed. With a quick look we would need to:
1. Generalize the EVM nodes connectivity logic per manager. -- should not be hard
2. Abstract away the etherscan part and use a chain agnostic etherscan. They have APIs for some evm chains. I think as long as there is a ***scan page available there is an API. Like for optimism: https://optimistic.etherscan.io/apis#accounts


Only if we achieve those would it make sense to rename and place in other directories. Now for a directory tree I assume something like:

```
evm/ --
         |
         |--manager.py
         |--scan.py
         |-- ethereum/ --
         |                      |
         |                      |--manager.py
         |                      |--scan.py
         |
         |--polygon/------
         |                      |
         |                      |--manager.py
         |                      |--scan.py
```

Where the top level manager.py and scan.py would be the interface/abstract class definition of how a manager and an ***scan class should be with all the common logic defined. And then in each subdirectory the specific chain?

Just some thoughts here.

**Comment by PabloCastellano** [comment]
@LefterisJP Yeah that makes sense. I have some of that already done in my local branch implementing Polygon support. Ok, since there are unrelated changes in this PR, I will split them in smaller PRs.

**Comment by PabloCastellano** [comment]
Hey @LefterisJP, I have just rebased this PR. It's much easier to review than it was initially after the smaller PRs already merged.

File: rotkehlchen/chain/manager.py [hunk 1]
```diff
@@ -1209,10 +1216,9 @@ def add_ethereum_lp_to_balance(
                 else:
                     balances[loc_key][EvmToken(ethaddress_to_identifier(lp.address))] = lp.user_balance  # noqa: E501
 
-    def _add_protocol_balances(self) -> None:
+    def _add_protocol_balances(self, balances: DefaultDict[ChecksumEvmAddress, BalanceSheet]) -> None:  # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/chain/manager.py:1219]: no no this is ethereum specific -- least for now. You should not rename it to `balances`

**Review by LefterisJP — APPROVED** [review]
I have left quite a few comments on this but since it's green and made by an external contributor I will merge and handle the comments on my own when I find some time.

Especially since the team is going to be touching lots of code around there and as such it's gonna get hard to keep external PRs up to date.

**Pull request merged**